core/modules/widgets/navigator.js

```diff
@@ -373,6 +410,10 @@ NavigatorWidget.prototype.handleCancelTiddlerEvent = function(event) {
 					this.removeTitleFromStory(storyList,draftTitle);
 				}
 				this.saveStoryList(storyList);
+				if(adjacentTiddler) {
+					this.wiki.setText(this.historyTitle,"previous-focus-tiddler",undefined,this.wiki.getTiddler(this.historyTitle).getFieldString("focus-tiddler"));
+					this.wiki.setText(this.historyTitle,"focus-tiddler",undefined,adjacentTiddler);
+				}
```

Comment:
these 2 lines are duplicated way too often. IMO that's a sign, that theres something wrong.

Reply:
@pmario, this is refactored now